### PR TITLE
feature/8942-ClaimFilesSegment

### PR DESCRIPTION
**Pull request by Sparowhawk**
## Description of Change
Created Files layout and segmented controller title change from details to files. Also created unit tests. This is behind the feature flag. See slack thread for VQA

## Screenshots/Video

Screenshot 2024-07-12 at 11 04 17 AM
Screenshot 2024-07-18 at 11 11 06 AM

## Testing
yarn test

- [x] Tested on iOS 
- [x] Tested on Android 

## Reviewer Validations
Use a row component to represent each file associated with claim

 Within the row component, display these additional fields associated with each file:

Filename
Request type
Document type
Received
 VQA Misty / Lauren

## PR Checklist

  **Reviewer:** Confirm the items below as you review

- [x] PR is connected to issue(s)
- [x] Tests are included to cover this change (when possible)
- [x] No magic strings (All string unions follow the [Union -> Constant](https://github.com/department-of-veterans-affairs/va-mobile-app/blob/develop/VAMobile/src/constants/common.ts) type pattern)
- [x] No secrets or API keys are checked in
- [x] All imports are absolute (no relative imports)
- [x] New functions and Redux work have proper TSDoc annotations

## For QA

[Run a build for this branch](https://github.com/department-of-veterans-affairs/va-mobile-app/actions/workflows/on_demand_build.yml)

### CODE REVIEW

**Review by alexandec — APPROVED** [review]

File: VAMobile/src/screens/BenefitsScreen/ClaimsScreen/ClaimDetailsScreen/ClaimDetailsScreen.test.tsx [hunk 1]
```diff
@@ -71,14 +75,31 @@ context('ClaimDetailsScreen', () => {
...
+        .mockResolvedValue({
+          data: {
+            ...claimData,
+          },
+        })
+      renderWithData(true, {
+        ...claimData,
+      })
+      await waitFor(() => fireEvent.press(screen.getByText('Files')))
+      await waitFor(() => fireEvent.press(screen.getByText('Files')))
```
Comment (alexandec) [inline VAMobile/src/screens/BenefitsScreen/ClaimsScreen/ClaimDetailsScreen/ClaimDetailsScreen.test.tsx:99]: Does this need to be pressed twice?

Comment (Sparowhawk) [reply VAMobile/src/screens/BenefitsScreen/ClaimsScreen/ClaimDetailsScreen/ClaimDetailsScreen.test.tsx:99]: unfortunately it did, just like the details tests above. I haven't figured out why

**Review by alexandec — DISMISSED** [review]

**Review by alexandec — APPROVED** [review]

File: VAMobile/src/screens/BenefitsScreen/ClaimsScreen/ClaimDetailsScreen/ClaimFiles/ClaimFiles.tsx [hunk 2]
```diff
@@ -16,23 +16,41 @@ function ClaimFiles({ claim }: ClaimFilesProps) {
   const { t } = useTranslation(NAMESPACE.COMMON)
   const theme = useTheme()
   const { attributes } = claim
-  const documents = attributes.eventsTimeline.filter((event) => event.filename && event.filename.length > 0)
+  const events = attributes.eventsTimeline.filter(
+    (event) => (event.filename && event.filename.length > 0) || (event.documents && event.documents.length > 0),
```
Comment (alexandec) [inline VAMobile/src/screens/BenefitsScreen/ClaimsScreen/ClaimDetailsScreen/ClaimFiles/ClaimFiles.tsx:20]: Optionally could simplify these checks like

```
(event.filename?.length || event.documents?.length)
```

File: VAMobile/src/screens/BenefitsScreen/ClaimsScreen/ClaimDetailsScreen/ClaimFiles/ClaimFiles.tsx [hunk 3]
```diff
@@ -16,23 +16,41 @@ function ClaimFiles({ claim }: ClaimFilesProps) {
...
+        if (event.uploadDate) {
+          textLines.push({ text: t('appointmentList.received', { date: event.uploadDate }) })
         }
         items.push({ textLines: textLines })
+      } else {
+        _.forEach(event.documents || [], (document) => {
+          if (document.filename) {
+            const textLines: TextLine[] = [{ text: document.filename, variant: 'MobileBodyBold' }]
+            if (document.fileType) {
+              textLines.push({ text: t('appointmentList.requestType', { type: document.fileType }) })
```
Comment (alexandec) [inline VAMobile/src/screens/BenefitsScreen/ClaimsScreen/ClaimDetailsScreen/ClaimFiles/ClaimFiles.tsx:43]: Optionally could simplify all these to one-liners like

```
document.fileType && textLines.push({ text: t('appointmentList.requestType', { type: document.fileType }) })
```

File: VAMobile/src/screens/BenefitsScreen/ClaimsScreen/ClaimDetailsScreen/ClaimFiles/ClaimFiles.tsx [hunk 4]
```diff
@@ -16,23 +16,41 @@ function ClaimFiles({ claim }: ClaimFilesProps) {
...
+            }
+            if (document.documentType) {
+              textLines.push({ text: t('appointmentList.documentType', { type: document.documentType }) })
+            }
+            if (document.uploadDate) {
+              textLines.push({ text: t('appointmentList.received', { date: document.uploadDate }) })
+            }
+            items.push({ textLines: textLines })
```
Comment (alexandec) [inline VAMobile/src/screens/BenefitsScreen/ClaimsScreen/ClaimDetailsScreen/ClaimFiles/ClaimFiles.tsx:51]: Could use shorthand here

**Review by alexandec — APPROVED** [review]
Approving with optional suggestions

**Review by rbontrager — APPROVED** [review]

**Review by rbontrager — APPROVED** [review]

**Review by dumathane — APPROVED** [review]

**Pull request merged**